log4j-api-test/src/main/java/org/apache/logging/log4j/test/junit/TempLoggingDirectory.java

```diff
@@ -135,6 +132,29 @@ private PathHolder createLoggingPath(final ExtensionContext context, final Clean
         return holder;
     }
 
+    private Path determinePerClassPath(ExtensionContext context) {
+        // Check if the parent context already created a folder
+        PathHolder holder = ExtensionContextAnchor.getAttribute(PathHolder.class, PathHolder.class, context);
+        if (holder == null) {
+            try {
+                // Create temporary per-class directory
+                final String baseDir = System.getProperty("basedir");
+                final Path basePath = (baseDir != null ? Paths.get(baseDir, "target") : Paths.get(".")).resolve("logs");
+                final Class<?> clazz = context.getRequiredTestClass();
+                final Package pkg = clazz.getPackage();
+                final String dir =
+                        pkg.getName().replaceAll("[.$]", File.separatorChar == '\\' ? "\\\\" : File.separator);
+                // Create a temporary directory that uses the simple class name as prefix
+                Path packagePath = basePath.resolve(dir);
+                Files.createDirectories(packagePath);
+                return Files.createTempDirectory(packagePath, clazz.getSimpleName());
```

Comment:
Can we suffix the path with something that will make directories lexicographically sorted by their creation time, please? For instance, `timestamp + counter + salt`, etc.

Reply:
I added a counter in https://github.com/apache/logging-log4j2/pull/2954/commits/463c5e8f0231f136685d5fb56c3cb418820b3157

Reply:
`Files.createTempDirectory()` already takes care of the distinctiveness – I just want to be able to spot the last effective folder. I don't see how adding just a counter helps, since counters get reset at reruns. Am I missing something?

Reply:
Aren't those reruns happening inside the same JVM?

**Note**: the directories are deleted by default if the test is successful, so we are only looking at failed runs. I wouldn't want to have too long names.